app/services/Countries.scala

```diff
@@ -45,7 +45,13 @@ object Countries {
 
   val countryCodeMap: Map[String, Country] = countries.map(country => (country.countryCode, country)).toMap
 
-  def retrieveCountryNameFromCode(code: String): Country = countryCodeMap(code)
+  val countryNameMap: Map[String, Country] = countries.map(country => (country.countryName, country)).toMap
+
+  def retrieveCountryFromCode(code: String): Country = countryCodeMap(code)
+
+  def retrieveCountryFromName(name: String): Country = countryNameMap(name)
```

Comment:
Similar

Reply:
Just interested.  Why do we need to look up by name?

Reply:
because our models are not align, somewhere we're storing code, somewhere whole name. Of course in places where we storing whole name we need to have code...

Changed